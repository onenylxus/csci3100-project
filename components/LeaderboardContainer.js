// Import
import React from 'react';
import { Image, Text, View } from 'react-native';
import { Col, Grid } from 'react-native-easy-grid';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faMedal } from '@fortawesome/free-solid-svg-icons';
import Style from '../assets/style';

// Export Leaderboard
export default function LeaderboardContainer({
  username,
  image,
  popularity,
  rank,
}) {
  // Medal and Ranking
  function showMedal(ranking) {
    if (ranking === 1) {
      return (
        <FontAwesomeIcon
          icon={faMedal}
          style={{ color: '#ffd700', marginLeft: 10 }}
          size={30}
        />
      );
    }
    if (ranking === 2) {
      return (
        <FontAwesomeIcon
          icon={faMedal}
          style={{ color: '#c0c0c0', marginLeft: 10 }}
          size={30}
        />
      );
    }
    if (ranking === 3) {
      return (
        <FontAwesomeIcon
          icon={faMedal}
          style={{ color: '#cd7f32', marginLeft: 10 }}
          size={30}
        />
      );
    }
    return <Text style={Style.rankText}>#{rank}</Text>;
  }

  return (
    <View style={Style.LeaderboardContainer}>
      <View style={{ flexDirection: 'row' }}>
        <Grid>
          <View style={{ width: '15%' }}>{showMedal(rank)}</View>
          <Col style={{ flexDirection: 'row', marginVertical: 20 }}>
            <Image
              style={Style.userIcon}
              source={
                image &&
                username !== 'Anonymous' &&
                username !== 'deleted account'
                  ? { uri: `data:image/jpeg;base64,${image}` }
                  : require('../assets/images/profile.png')
              }
            />
            <View style={{ flexDirection: 'column' }}>
              <Text style={{ fontSize: 18, fontWeight: 'bold' }}>
                {username}
              </Text>
              <Text>Popularity: {popularity}</Text>
            </View>
          </Col>
        </Grid>
      </View>
    </View>
  );
}
