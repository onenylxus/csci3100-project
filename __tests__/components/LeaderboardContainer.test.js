// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import LeaderboardContainer from '../../components/LeaderboardContainer';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('Comment', () => {
  let element;

  beforeEach(() => {
    // Mock username
    const usernameMock = 'testac';
    const popularityMock = 0;

    // Render
    element = render(<LeaderboardContainer username={usernameMock} popularity={popularityMock} />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
