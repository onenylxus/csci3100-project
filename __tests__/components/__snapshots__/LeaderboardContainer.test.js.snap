// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment matches snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#f8f8f8",
      "borderRadius": 15,
      "margin": 4,
      "width": "85%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "width": "15%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "flex-start",
              "fontFamily": "Roboto",
              "fontSize": 15,
              "marginLeft": 13,
              "marginVertical": 5,
            }
          }
        >
          #
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 28,
              "height": 32,
              "marginHorizontal": 8,
              "marginTop": 4,
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            testac
          </Text>
          <Text>
            Popularity: 
            0
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
